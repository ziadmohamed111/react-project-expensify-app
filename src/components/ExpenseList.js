import React from 'react'
import { connect } from "react-redux"
import ExpenseListItem from './ExpenseListItem'
import selectExpense from "../selectors/expenses"

export const ExpenseList = ( props ) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        {
            props.expenses.length === 0 ? (
                <div className="list-item list-item--message">
                    <span>No expenses</span>
                </div>
            ) : ( 
                props.expenses.map( ( expense ) => <ExpenseListItem key={expense.id} {...expense} /> ) 
                )

    }
    </div>
);
const mapStateIntoProps = ( state ) => {
    return {
        expenses: selectExpense( state.expenses , state.filters )
    }
}

export default connect( mapStateIntoProps )( ExpenseList )
