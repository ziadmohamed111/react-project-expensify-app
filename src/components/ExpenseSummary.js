import React from 'react';
import {connect} from "react-redux"
import {Link} from "react-router-dom"
import totalExpenses from "./totalExpenses"
import numeral from "numeral"
import selectExpenses from "../selectors/expenses"

export const ExpenseSummary = ( { expensesCount, expenseTotal}) => {
    const expenseWord = expensesCount === 1 ? "expense" : "expenses" 
    const formatedTotalExpenses = numeral( expenseTotal / 100 ).format( '$0,0.00' )
    return(
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header_title">Viewing <span> {expensesCount}</span> {expenseWord} totalling <span> {formatedTotalExpenses} </span> </h1>
                <div className="page-header_actions">
                    <Link className="button-big" to="/create">Add Expense</Link>
                </div>
            </div>
     
        </div>
    )

}

const mapStateToProps = ( state ) => {
    const visExpenses = selectExpenses(state.expenses , state.filters)
    return {
        expensesCount : visExpenses.length,
        expenseTotal: totalExpenses(visExpenses)
    }
}

export default connect( mapStateToProps )( ExpenseSummary)