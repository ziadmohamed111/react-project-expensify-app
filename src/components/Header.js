import React from "react";
import { Link } from "react-router-dom"
import {startLogout} from "../actions/auth"
import {connect} from "react-redux"

export const Header = (props) => (
    <header className="header">
        <div className="content-container">
            <div className="header_content">
                <Link className="header__title" to="/dashboard">  
                    <h1>Expensify</h1>
                </Link>
                <button className="button-inline" onClick={props.startLogout}>Logout</button>
            </div>
        </div>
    </header>
)
const mapDispatchToProps = (dispatch) =>({
    startLogout: () => dispatch( startLogout())
})
export default connect(undefined , mapDispatchToProps)(Header)