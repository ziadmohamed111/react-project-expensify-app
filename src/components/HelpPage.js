import React from 'react';

export const HelpPage = () => (
    <div>
        <p>HelpPage</p>
    </div>

)

export default HelpPage