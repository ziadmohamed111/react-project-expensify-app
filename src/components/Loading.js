import React from "react";

const Loading = ()=>(
    <div className="Loading-Page">
        <div className="loading">
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
            <div className="loading-div"></div>
        </div>
    </div>
)
export default Loading