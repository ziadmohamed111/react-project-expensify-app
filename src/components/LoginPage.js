import React from "react"
import { connect } from "react-redux"
import { startLogin } from "../actions/auth"

export const LoginPage = ( props ) => (
    <div className="box-layout">
        <div className="box-layout__login-box">
            <h1>Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button-big" onClick={props.startLogin}>Login With Google</button>
        </div>
        
    </div>
)

const mapDispatchToProps = ( dispatch ) => ( {
    startLogin: () => dispatch( startLogin() )
} )

export default connect( undefined, mapDispatchToProps )( LoginPage ) 