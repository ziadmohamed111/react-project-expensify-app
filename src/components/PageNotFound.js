import React from 'react';
import { Link } from "react-router-dom" 

export const PageNotFound = () => (
    <div>
        <p>404!</p>
        <Link to="/">GO HOME</Link>
    </div>

)

export default PageNotFound