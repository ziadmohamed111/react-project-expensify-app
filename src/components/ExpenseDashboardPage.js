import React from 'react';
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"

export const ExpenseDashboardPage = () => {    
    return (
        (
            <div>
                <ExpenseListFilters/>
                <ExpenseList/>
            </div>
        )
    )
}

export default ExpenseDashboardPage