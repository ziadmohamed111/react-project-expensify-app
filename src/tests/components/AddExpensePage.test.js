import React from "react"
import { shallow } from "enzyme"
import { AddExpensePage } from "../../components/AddExpensePage"
import expenses from "../fixtures/expenses"

let history, startAddExpense, wrapper
beforeEach( () => {
    startAddExpense = jest.fn()
    history = {
        push: jest.fn()
    }
    wrapper = shallow( <AddExpensePage startAddExpense={startAddExpense} history={history} /> )
} )

test( "should render Add Expense Page Correctly", () => {
    expect( wrapper ).toMatchSnapshot()
} )

test( "should handel on Submit", () => {
    wrapper.find( "ExpenseForm" ).prop( "onSubmit" )( expenses[1] )
    expect( history.push ).toHaveBeenLastCalledWith( "/" )
    expect( startAddExpense ).toHaveBeenLastCalledWith( expenses[1] )
} )