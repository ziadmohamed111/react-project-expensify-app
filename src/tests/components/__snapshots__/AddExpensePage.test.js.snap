// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Add Expense Page Correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header_title"
      >
        Add Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
